core/src/server/helpers.rs

```diff
@@ -24,20 +24,67 @@
 // IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
+use std::borrow::Cow as StdCow;
 use std::io;
 use std::time::Duration;
 
 use jsonrpsee_types::error::{
 	reject_too_big_batch_response, ErrorCode, ErrorObject, OVERSIZED_RESPONSE_CODE, OVERSIZED_RESPONSE_MSG,
 };
-use jsonrpsee_types::{Id, InvalidRequest, Response, ResponsePayload};
+use jsonrpsee_types::{ErrorObjectOwned, Id, InvalidRequest, Response, ResponsePayload};
 use serde::Serialize;
 use serde_json::value::to_raw_value;
 use tokio::sync::mpsc;
 
+use super::{DisconnectError, SendTimeoutError, SubscriptionMessage, TrySendError};
 use crate::server::LOG_TARGET;
 
-use super::{DisconnectError, SendTimeoutError, SubscriptionMessage, TrySendError};
+/// Wrapper over [`tokio::mpsc::oneshot::channel`]..
+pub fn response_channel() -> (MethodResponseNotifyTx, MethodResponseNotifyRx) {
+	let (tx, rx) = tokio::sync::oneshot::channel();
+	(tx, MethodResponseNotifyRx(rx))
+}
+
+/// ..
+pub type MethodResponseNotifyTx = tokio::sync::oneshot::Sender<NotifyMsg>;
```

Comment:
nit: We could wrap this into a struct to avoid breaking changes in user-code; although they are not supposed to use it